### ISSUE TITLE
Zugriff über SSL

### ISSUE THREAD

**Issue by John-Rookie** [JsBergbau/BaikalAnleitung#16]
Hallo.
Nachdem nun nach 7 Jahren meine SD-Karte im Raspi defekt ist musste ich mein Baikal erneuern. Dieses hatte ein damaliger Freund für mich eingerichtet. Ich will also damit sagen, dass ich keine Ahnung habe.
Deshalb erstmal vielen Dank für die **fantastische Anleitung**, denn damit habe ich es als absoluter Laie zum laufen bekommen.
NUR. Ich habe versucht die baikal-ngnix.site duch kommentieren und auskommentieren auf SSL Zugang umzustellen. Ist mir leider nicht gelungen. Der Webserver läuft dann garnicht. Ich wäre über eine funktionierende Konfiguration dankbar oder eine, auch so fantastische Anleitung dazu.

**Comment by Bl8Runnr** [comment]
Soweit ich mich erinnere, musst du im Root deiner SSD eine Datei anlegen mit dem Namen ssh (Ohne Dateiendung, ohne Inhalt).
Danach solltest du zugreifen können.

**Comment by JsBergbau** [comment]
Hallo John-Rookie,

vielen Dank für dein Lob. Das freut mich besonders, wenn du sagst, dass du es als absoluter Laie zum Laufen bekommen hast.

Wenn du den Zugang via SSL realisieren möchtest, dann entnehme ich dem, dass du den Server frei zugänglich ins Internet hängen möchtest. Das ist mit diversen Gefahren verbunden und wenn man regelmäßig zu Hause ist, reicht auch der Zugang vom heimischen WLAN vollkommen aus und da braucht es keine SSL-Verschlüsselung.

Wenn du dennoch unbedingt SSL haben möchtest, dann poste doch mal eine funktionsfähige, deine nicht mehr funktionsfähige Konfiguration, sowie die Fehlermeldung, insbesondere die unter /var/log/ningx/*error.log hier.

**Comment by John-Rookie** [comment]
Ja, der gefahren bin ich mir durchaus bewusst, möchte es aber vor allem wegen der bisherigen Gerätekonfigurationen wieder nutzen. Ich dachte da kann ich mir ersparen alle Sync-Geräte zu modifizieren. Hab auch ein Update. nginx läuft mit SSL Zugriff und dieser Konfiguration:
`server {

    # listen 9999 default_server;

    #listen [::]:80 default_server;
    # SSL configuration
    #
    listen 443 ssl default_server;
    # listen [::]:443 ssl default_server;
    #
    # Note: You should disable gzip for SSL traffic.
    # See: https://bugs.debian.org/773332
    #
    # Read up on ssl_ciphers to ensure a secure configuration.
    # See: https://bugs.debian.org/765782
    #
    # Self signed certs generated by the ssl-cert package
    # Don't use them in a production server!
    #
    include snippets/snakeoil.conf;
    root /var/www/html;

    # root /home/pi/DiensteUndProgramme/Baikal/html;

    # Add index.php to the list if you are using PHP

    index index.php index.html index.htm index.nginx-debian.html;
    server_name _;

    # location / {
    #     First attempt to serve request as file, then
    #     as directory, then fall back to displaying a 404. try_files $uri $uri/ =404;
    # }

    rewrite ^/.well-known/caldav /dav.php redirect;
    rewrite ^/.well-known/carddav /dav.php redirect;
    charset utf-8;

    location ~ /(\.ht|Core|Specific|config) {
        deny all;
        return 404;
    } #In der aktuellen Variante nicht benötigt, da "html" das Root Verzeichnis ist, schadet aber auch nicht.
	 #Sollte man Baikal jedoch direkt im Webserver als Unterverzeichnis installieren, absolut notwendig. Bitte daran denken!

    location ~ ^(.+.php)(.*)$ {
        try_files $fastcgi_script_name =404;
        include /etc/nginx/fastcgi_params;
        fastcgi_split_path_info ^(.+.php)(.*)$;
    #fastcgi_pass unix:/run/php/php7.3-fpm.sock; #Hier auf eure installierte PHP-Version achten. 
	    fastcgi_pass unix:/run/php/php7.4-fpm.sock; #Ubuntu hat z.B. schon php 7.4
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param PATH_INFO $fastcgi_path_info;
    }

    # deny access to .htaccess files, if Apache's document root
    # concurs with nginx's one
    #

    location ~ /.ht {
        deny all;
    }

}`
Nur auf die Baikal Webseite geht es nicht weiter.

**Comment by JsBergbau** [comment]
> Nur auf die Baikal Webseite geht es nicht weiter.

Welche Fehlermeldung wird angezeigt?

**Comment by John-Rookie** [comment]
https://<IP>/baikal bringt Fehler 404 Not Found.
Im Protokol steht:

2022/02/06 14:35:00 [error] 1156#1156: *1 open () "/var/www/html/baikal" failed (2: No such file or directory), client: 192.168.0.148, server: _, request: "GET /baikal HTTP/1.1", host: "192.168.0.184"

**Comment by JsBergbau** [comment]
```
root /var/www/html;
# root /home/pi/DiensteUndProgramme/Baikal/html;
```
Ich gehe mal davon aus, dass Baikal wie in der Anleitung angegeben unter /home/pi/DiensteUndProgramme/Baikal abgelegt wurde.
Daher obiges zu 
```
# root /var/www/html;
root /home/pi/DiensteUndProgramme/Baikal/html;
```
ändern. Anschließend sollte https://192.168.0.184/ funktionieren. 

Wenn es unter https://192.168.0.184/baikal zur Verfügung gestellt werden soll, kannst du dir mal https://serverfault.com/questions/778329/nginx-php-in-a-subdirectory ansehen. Das könnte das sein, was du in diesem Fall brauchst.

**Comment by John-Rookie** [comment]
Danke. So habe ich erstmal Zugriff. Das mit dem subdirectorys werde ich mir mal ansehen.
Wenn nicht, kann ich auch so damit arbeiten.
